test/integration/array_test.ts

```diff
@@ -108,6 +108,27 @@ describe('Array', function () {
     });
   });
 
+  it('can insert an element at the first of array', function () {
+    const doc = DocumentReplica.create('test-col', 'test-doc');
+    assert.equal('{}', doc.toSortedJSON());
+
+    doc.update((root) => {
+      root['list'] = [0, 1, 2];
+    }, 'set {"list":[0,1,2]}');
+
+    doc.update((root) => {
+      const item = root['list'].getElementByIndex(2);
+      root['list'].moveFront(item.getID());
+      assert.equal('{"list":[2,0,1]}', root.toJSON());
+    });
+
+    doc.update((root) => {
+      const item = root['list'].getElementByIndex(1);
+      root['list'].moveFront(item.getID());
+      assert.equal('{"list":[0,2,1]}', root.toJSON());
+    });
+  });
+
```

Comment:
Since this test does not connect to Agent, how about moving this to `unit/document/document_test.ts`?

Reply:
sure! let me move this test code to document_test.ts